src/multisig/multisig.service.ts

```diff
@@ -0,0 +1,51 @@
+import { LoggerService } from '@/lib/logging/logger.service';
+import { Injectable } from '@nestjs/common';
+import * as colors from 'colors';
+import { InquirerService } from 'nest-commander';
+import { MultisigAnswers } from './multisig.questions';
+import { MULTISIG_QUESTION_SET_NAME } from './multisig.questions.descriptor';
+import { Network } from './multisig.safe.constants';
+import { NetworkService } from './network.service';
+import { SafeService } from './safe.service';
+
+@Injectable()
+export class MultisigService {
+    constructor(
+        private readonly logger: LoggerService,
+        private readonly inquirer: InquirerService,
+        private readonly networkService: NetworkService,
+        private readonly safeService: SafeService
+    ) {}
+
+    async runMultisigFlow(): Promise<string> {
+        const chainId = await this.networkService.askToChooseNetwork();
+
+        let multisigAddress = '';
+
+        if (chainId != Network.Skip) {
+            multisigAddress = await this.askForMultisigAddress();
+
+            if (multisigAddress) {
+                if (chainId != Network.Other) {
+                    await this.safeService.validateMultisigAddress(chainId, multisigAddress);
+                } else {
+                    this.logger.warn(
+                        `Network is set to "Other". Skipping multisig address validation using ${colors.cyan(
+                            'SAFE'
+                        )} service`
+                    );
+                }
+            }
+        }
+
+        multisigAddress
+            ? this.logger.log(`Multisig address provided: ${colors.cyan(multisigAddress)}`)
+            : this.logger.warn('Multisig address is not provided or invalid. Skipping multisig support.');
+
+        return multisigAddress;
```

Comment:
Suggesting a little bit of clean up for the implementation:

```typescript
async getMultiSigAddress(): Promise<string | undefined> {
        const chainId = await this.networkService.askToChooseNetwork();
        
        if (chainId === Network.Skip) {
            this.logger.warn(`Skipping Multi-Sig integration`);
        }
        else {
            const multisigAddress = await this.askForMultisigAddress();
            this.logger.log(`Using Multi-Sig Address: ${colors.cyan(multisigAddress)}`);
            
            if (chainId !== Network.Other) {
                this.logger.log(`Validating Multi-Sig Address on ${colors.cyan(Network[chainId])} network`);
                await this.safeService.validateMultisigAddress(chainId, multisigAddress);
            }

            return multisigAddress;
        }
    }
```

> note that the function return value was updated to `Promise<string | undefined>` _(instead of returning an empty string)_.

Reply:
agree, like ur suggestion